src/Marten/Linq/Parsing/SimpleBinaryComparisonExpressionParser.cs

```diff
@@ -63,7 +63,7 @@ public IWhereFragment Parse(IQueryableDocument mapping, ISerializer serializer,
 
             var useContainment = mapping.PropertySearching == PropertySearching.ContainmentOperator || field.ShouldUseContainmentOperator();
 
-            var isDuplicated = (mapping.FieldFor(members) is DuplicatedField);
+            var isDuplicated = field is DuplicatedField || field is IdField;
```

Comment:
Suggestion: Rather than including `|| field is IdField` in the `isDuplicated`. It would be better to have a seperate variable `var isIdField = field is IdField;` and use it in the further checks to add clarity.

Reply:
Jeremy was okay with the PR, closing this suggestion as resolved.